### PR TITLE
[react-select] Add data to MultiValueRemoveProps

### LINKED ISSUE #50348: [@types/react-select]: MultiValueRemove missing data props
`multiValueRemove` / `MultiValueRemove` (specifically `MultiValueRemoveProps`) receives `data` just like the other components

### tsc

```typescript
[tsl] ERROR in src/react-select.tsx(149,32)
      TS2339: Property 'data' does not exist on type 'CommonProps<IOptionType, true> &
      { children: ReactNode; innerProps: { className: string; onTouchEnd: (event: any) => void; 
      onClick: (event: any) => void; onMouseDown: (event: any) => void; }; selectProps: any; }'.
```

### See react-select:

`MultiValueRemove` uses `MultiValueGenericProps` (which has `data`) consistently since 2.0:

- 3.0.0 (Apr 29, 2019), [`MultiValueRemove`](https://github.com/JedWatson/react-select/blob/3.0.0/docs/PropTypes/components/MultiValueRemove.js) + [`MultiValueGenericProps`](https://github.com/JedWatson/react-select/blob/v3.0.0/packages/react-select/src/components/MultiValue.js#L64)
- 2.0.0 (Jul 22nd, 2018): [`MultiValueRemove`](https://github.com/JedWatson/react-select/blob/v2.0.0/docs/PropTypes/components/MultiValueRemove.js) + ([`MultiValueGenericProps`](https://github.com/JedWatson/react-select/blob/v2.0.0/src/components/MultiValue.js#L54s)

Official example of `multiValueRemove` receiving `{ data }`: https://github.com/JedWatson/react-select/blob/2f2aca2/docs/examples/StyledMulti.js#L46-L53

For 1.x `multiValueRemove` didn't get data props it seems: https://github.com/JedWatson/react-select/blob/451bcd5/docs/PropTypes/components/MultiValueRemove.js

### @types/react-select

As of December 1st / 3.1.1, `MultiValueRemove` doesn't get the `data` types

The reason why is it doesn't use `MultiValueGeneric` like the other customizable components do

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e52c6aa/types/react-select/src/components/MultiValue.d.ts#L45

### Ramifications

I style based off the value of the data (just like any other component being styled)

### Rest

- [x] I tried using the `@types/react-select` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @claasahl @jonfreedman @Methuselah96 @thchia @danieldelcore @rjoonas @kylemh 

If you do not mention the authors the issue will be ignored.

### PR DESCRIPTION
**Pull request by Methuselah96**
Resolves https://github.com/DefinitelyTyped/DefinitelyTyped/issues/50348.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Methuselah96/react-select/blob/typescript-final/packages/react-select/src/components/MultiValue.tsx#L134
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## redux-orm ([<kbd>unpkg</kbd>](https://unpkg.com/browse/redux-orm@latest/))
was missing the following properties:
1. Session
1. Schema
1. Backend
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ecb55a83bcc9204e44f24234fd37d0a12e667e42
</p>

File: types/react-select/src/components/MultiValue.d.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export const MultiValueContainer: typeof MultiValueGeneric;
 export const MultiValueLabel: typeof MultiValueGeneric;
 export type MultiValueRemoveProps<OptionType extends OptionTypeBase> = CommonProps<OptionType, true> & {
   children: ReactNode,
+  data: OptionType,
```
Comment (tony) [inline types/react-select/src/components/MultiValue.d.ts:37]: Looks good (as do the other props)

https://github.com/JedWatson/react-select/blob/v3.0.0/packages/react-select/src/components/MultiValue.js#L48

I notice we use `data: OptionType` throughout (while `data: any` seems to be used in react-select)

I assume that's okay and just additional typesafety for us 👍

Comment (Methuselah96) [reply types/react-select/src/components/MultiValue.d.ts:37]: Yup.

**Review by kylemh — APPROVED** [review]

**Comment by Methuselah96** [comment]
Ready to merge

**Pull request merged**